### DISCUSSION TITLE
Is it normal that sync checks every file / syncing a single subdir doesn't work

### DISCUSSION THREAD

**Discussion by nwum** [abraunegg/onedrive#2236] [Q&A] [answered]
I have like 200GB on my OneDrive and --verbose tells me that syncing checks every single file when 99% of them are unchanged. That combined with it having to "process to ensure consistent local state due to a full scan being requested", every sync takes a while especially compared to my OneDrive experience on Windows.
If that's normal and there's no way around it then so be it. However, I've also tried syncing only a single subdirectory to make it faster but I've kept getting "Remote change discarded - not in --single-directory sync scope (in DB)" messages in verbose.
Thanks in advance.

**Comment by abraunegg** [comment, marked as answer]
@nwum 
What application version are you using? What is the output of `onedrive --version` as this is missing from your post.

> I have like 200GB on my OneDrive and --verbose tells me that syncing checks every single file when 99% of them are unchanged.

Given that you are running verbose - you should also see that when it is checking these files it is to ensure their data and integrity.

> That combined with it having to "process to ensure consistent local state due to a full scan being requested", every sync takes a while especially compared to my OneDrive experience on Windows.

This is really dependent on 'how' you are running the application. If you are using `--synchronize` this is a *one-time-sync* .. thus, everything needs to be checked each and every time you run the application.

If you are using `--monitor` the full scan is done once at application startup, then, only done on a rotational basis governed by these two configuration values:
* monitor_interval - https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#monitor_interval
* monitor_fullscan_frequency - https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#monitor_fullscan_frequency

> If that's normal and there's no way around it then so be it. However, I've also tried syncing only a single subdirectory to make it faster but I've kept getting "Remote change discarded - not in --single-directory sync scope (in DB)" messages in verbose.

Syncing single directories should be faster, however that message is correct. Now you are thinking *why* is this message correct ... well - OneDrive as an application was written by Microsoft with the major assumption that what is stored online, is what the user wants to have local - however, this client allows 'client side filtering' (like the use of `--single-directory` and others) - however the OneDrive API has zero capability to tell OneDrive our sync directives - it sends *all* data to the client - so we have to filter the data from OneDrive - hence the discarded message.

**Reply by nwum** [reply]
I'm running onedrive v2.4.21

Okay so what I take from this is that there is no way around checking everything at once. I still have a few questions if that doesn't bother you:
1. Is --monitor just --synchronize happening every x seconds?
2. Should I add `onedrive --monitor `to my initrc if I want it to synchronize in the background without taking up a terminal?
Thanks for the speedy response.

**Reply by abraunegg** [reply]
> Is --monitor just --synchronize happening every x seconds?

Not exactly. Monitor 'monitors' your local file system for any changed files. Once a change has been detected, the updated local file is uploaded to OneDrive immediately. Changes are checked on OneDrive online based on 'monitor_interval'. The 'integrity' check is governed by 'monitor_fullscan_frequency'

> Should I add onedrive --monitor to my initrc if I want it to synchronize in the background without taking up a terminal

If this is the correct approach for your platform then yes. Essentially it is more desirable to run the application as a system service and using `--monitor`, rather than performing manual syncs.
